R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,717 @@
+    "    }\n",
+    "    return(df)\n",
+    "}\n",
+    "print(\"Function 'analyzeImage' defined.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Iterate over the dataset and call run the analyze function for each of the images:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datasetId <- REPLACE_WITH_DATASET_ID\n",
+    "\n",
+    "channelName <- 'CDK5RAP2-C'\n",
+    "\n",
+    "dataset <- loadObject(server, \"DatasetData\", datasetId)\n",
+    "\n",
+    "# Keep the channel name, image name, image id, area, perimeter, and diameter of the largest ROIs\n",
+    "result <- data.frame(Channel = c('remove'), ImageName = c('remove'), Image = c(0), ROIIndex = c(0), Area = c(0), Perimeter = c(0), Diameter = c(0), stringsAsFactors = FALSE)\n",
+    "\n",
+    "images <- getImages(dataset)\n",
+    "for (image in images) {\n",
+    "    result <- tryCatch({\n",
+    "        analyzeImage(image, channelName, result, saveROIs = TRUE)\n",
+    "    }, warning = function(war) {\n",
+    "        message(paste(\"WARNING:  \", image@dataobject$getId(),war))\n",
+    "        return(result)\n",
+    "    }, error = function(err) {\n",
+    "        message(paste(\"ERROR:  \", image@dataobject$getId() ,err))\n",
+    "        return(result)\n",
+    "    }, finally = {\n",
+    "    })\n",
+    "}\n",
+    "\n",
+    "result <- result[-1,]\n",
+    "rownames(result) <- c()\n",
+    "\n",
+    "# set the correct datatypes\n",
+    "result$Channel <- as.factor(result$Channel)\n",
+    "result$Area <- as.numeric(result$Area)\n",
+    "result$Perimeter <- as.numeric(result$Perimeter)\n",
+    "result$Diameter <- as.numeric(result$Diameter)\n",
+    "\n",
+    "result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Open some of the other images of the dataset in the full viewer to check the ROIs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Optional Exercise:** The measurements (Area, Perimeter and Diameter) are currently showing the number of pixels. Find out the pixel size in nanometer and apply it accordingly. Hint: The pixel size is the same for all images. Select one image in OMERO.web and check its metadata (right hand side panel). Solution: See bottom."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pxSizeInNM <- NA # REPLACE_WITH_PIXEL_SIZE\n",
+    "if (!is.na(pxSizeInNM))\n",
+    "    # Replace result$Area, Perimeter, Diameter with the\n",
+    "    # size in nanometer, using the variable 'pxSizeInNM'\n",
+    "    # declared above.\n",
+    "    \n",
+    "    # Enter code here...\n",
+    "    \n",
+    "    # Just print out the data frame again to check result:\n",
+    "    result\n",
+    "}"
```

Comment:
```
Error in parse(text = x, srcfile = src): <text>:11:1: unexpected '}'
10:     result
11: }
    ^
Traceback:
```

That is what I get if I replace nothing and just execute the cell as it is.
Was that the idea ? I thought it would smoothly pass saying "whatever, you did not bother to do the exercise", instead of erroring ?